CONTRIBUTING.md

```diff
@@ -0,0 +1,125 @@
+
+## Developer Certificate of Origin
+
+To contribute to this project, you must agree to the Developer Certificate of
+Origin (DCO) for each commit you make. The DCO is a simple statement that you,
+as a contributor, have the legal right to make the contribution.
+
+See the [DCO](DCO) file for the full text of what you must agree to.
+
+The most common way to signify your agreement to the DCO is to add a signoff to
+the git commit message, using the `-s` flag of the `git commit` command. It will
+add a signoff the looks like following line to your commit message:
+
+```txt
+Signed-off-by: John Smith <john.smith@example.com>
+```
+
+You must use your real name and a reachable email address in the signoff.
+
+Alternately, instead of commits signoff, you can also leave a comment on the PR
+with the following statement:
+
+> I agree to the DCO for all the commits in this PR.
+
+Note that this option still requires that your commits are made under your real
+name and a reachable email address.
+
+## Contributing Code
+
+### Finding Issues
+
+Generally, pull requests that address and fix existing GitHub issues are assigned
+higher priority over those that don't. Use the existing issue labels to help you
+identify relevant and interesting issues.
+
+If you think something ought to be fixed but none of the existing issues
+sufficiently address the problem, feel free to
+[raise a new one](https://github.com/kanisterio/kanister/issues/new/choose).
+
+For new contributors, we encourage you to tackle issues that are labeled as
+`good first issues`.
+
+Regardless of your familiarity with this project, documentation help is always
+appreciated.
+
+Once you found an issue that interests you, post a comment to the issue, asking
+the maintainers to assign it to you.
+
+### Coding Standard
+
+In this project, we adhere to the style and best practices established by the Go
+project and its community.
+
+Specifically, this means:
+
+* adhering to guidelines found in the [Effective Go](https://go.dev/doc/effective_go) document
+* following the common [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+
+The [golangci-lint](https://golangci-lint.run/) tool is used to enforce many
+styling and safety rules.
+
+### Creating A Local Build
+
+See the [BUILD.md](BUILD.md) document for instructions on how to build, test and
+run Kanister locally.
+
+### Commit messages
+
+The basic idea is that we ask all contributors to practice
+[good git commit hygiene](https://www.futurelearn.com/info/blog/telling-stories-with-your-git-history)
+to make reviews and retrospection easy. Use your git commits to provide context
+for the reviewers, and the folks who will be reading the codebase in the months
+and years to come.
+
+Finalized commit messages should look similar to the following format:
+
+```text
+Short one line title
+
+An explanation of the problem, providing context, and why the change is being
+made.
+```
+
+### Submitting Pull Requests
```

Comment:
Hi @ihcsim 
is it a good idea to ask people to confirm with maintainers if the pull request that they are trying to raise would be a good addition. In some cases there are chances that the PR is raised but the maintainers don't think that the PR should get in.

Reply:
Absolutely. Ideally, we would have prior discussion in GitHub issues/discussions, before pull requests are submitted. That's why in this paragraph I directed contributors to look up GitHub issues first.

Reply:
I think it is okay to determine this on the PR itself. At least that's what I have seen on some other OSS repos